### PR TITLE
fixed header

### LINKED ISSUE #1470: add horizontal scroll for the dataTable
Having the table as scrollable rather than the whole page/report

### PR DESCRIPTION
**Pull request by Prahitha**
should fix #1470 
**Note:** This issue was initially created to have the headline (title of the page) fixed as the table scrolls.

Using `scrollX: true` changes the position of the scrollbar to the end of the table instead of the window -- which I believe we do not want. So I tried using `position: fixed` for the headline but then the column headers overlap with it (could be because the header is floating?). I then added some padding to the table wrapper but there seems to be some flickering when scrolling because of this. Please correct me if I'm wrong and any suggestions on how to fix this are welcome. Thanks!

https://user-images.githubusercontent.com/44160152/118875617-e3a4ba80-b909-11eb-8bf4-4df3dca32f90.mp4

### CODE REVIEW

**Comment by mglb** [comment]
`position: fixed` won't work here, as it pulls the element from document flow (not OK) and positions it relative to the viewport (OK on X axis, not OK on Y axis). Use `position: sticky` with fixed `left` coordinate and suitable width instead.

Note that `body` element (header's parent element) does not grow with the table (its width is equal to viewport width minus the margins), and `sticky` positioning works so that:

> The element is positioned according to the normal flow of the document, and then offset relative to its nearest scrolling ancestor **and containing block (nearest block-level ancestor)**
([source: MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/position#sticky))

You'll have to make `body` width equal to its content width with something like:
```css
width: max-content;
min-width: 100%;
box-sizing: border-box;
```

**Comment by Prahitha** [comment]
This is how it looks like now, please let me know if I have to change something. Thanks!
https://user-images.githubusercontent.com/44160152/120112025-c5547f80-c191-11eb-98eb-c73a6875ea3f.mp4

**Review by tgorochowik — COMMENTED** [review]
The outcome looks good I think - I only have one question - when scrolling right, shouldn't the tool names disappear once their content is not visible? I mean - shouldn't the header for the tag name be on top (I think it is there but it does not have any text currently)

**Comment by Prahitha** [comment]
yes, that's correct. I actually tried adding a classname for those 2 cells and keeping them fixed but that didn't work very well. I'll try a little more today and see how it goes.

**Review by mglb — CHANGES_REQUESTED** [review]
It would be nice to also "lock horizontally" the footer lines, i.e.:

```
Showing 1 to 325 of 325 entries
Download a summary in csv
Generated using sv-tests, revision: d5e09d1.
```

File: conf/report/report.css [hunk 1]
```diff
@@ -1,16 +1,33 @@
...
+    left: 32px;
+    width: calc(100vw - 64px);
 }
 
 .headline a {  /* Only subtly hint that this is a link */
     color: #00006f;
     text-decoration: none;
 }
 
+#headline {
```
[the remark this answers is not in the document]
Comment (Prahitha) [reply conf/report/report.css:20]: yes, i think this can be removed

**Comment by Prahitha** [comment]
- locked the lines in the footer horizontally
- the tool names now disappear when their content is not visible

File: conf/report/report.css [hunk 2]
```diff
@@ -1,16 +1,33 @@
...
+  position: fixed;
+  width: 100%;
+  text-align: center;
+}
+
+.dataTables_filter {
+  position: sticky;
+  right: 5%;
```
[the remark this answers is not in the document]
Comment (Prahitha) [reply conf/report/report.css]: thanks for the hint!

**Comment by mglb** [comment]
FYI:  headers of invisible columns in floating header are still visible in Firefox. Not sure why.
Screenshot_20210611_144734

**Comment by Prahitha** [comment]
Hmm... It seemed fine on mine though? (I only have 4 columns for now so I had to stretch them)
Screenshot (69)

**Comment by mglb** [comment]
This one works for me too; the one that appears when you scroll down doesn't.

**Comment by Prahitha** [comment]
Ah right, I only checked on Chrome and it seemed to work fine. Thanks for pointing that out!

**Review by tgorochowik — APPROVED** [review]
The issue in firefox is minor and in general this PR provides some visible improvements.

@Prahitha please create an issue with the description (and screenshots) of the issue with firefox

and let's merge this!

**Pull request merged**